platform/platform-configuration-listener/src/main/java/org/codice/ddf/configuration/store/SystemConfigurationMigrator.java

```diff
@@ -0,0 +1,193 @@
+        exportCrl(exportDirectory);
+        exportPdpPolicies(exportDirectory);
+    }
+
+    private void exportSystemFiles(Path exportDirectory) throws MigrationException {
+        exportSystemProperties(exportDirectory);
+        exportUsersProperties(exportDirectory);
+    }
+
+    private void exportKeystores(Path exportDirectory) throws MigrationException {
+        exportKeystore(exportDirectory);
+        exportTruststore(exportDirectory);
+    }
+
+    private void exportKeystore(Path exportDirectory) throws MigrationException {
+        String keystore = getProperty(KEYSTORE_SYSTEM_PROP);
+        if (StringUtils.isBlank(keystore)) {
+            throw new MigrationException(String.format(
+                    "Cannot find keystore. System property %s is not set.", KEYSTORE_SYSTEM_PROP));
+        }
+        Path source = ddfHome.resolve(keystore);
+        Path destination = constructDestination(source, exportDirectory);
+        copyFile(source, destination);
+    }
+
+    private void exportTruststore(Path exportDirectory) throws MigrationException {
+        String truststore = getProperty(TRUSTSTORE_SYSTEM_PROP);
+        if (StringUtils.isBlank(truststore)) {
+            throw new MigrationException(String.format(
+                    "Cannot find truststore. System property %s is not set.",
+                    TRUSTSTORE_SYSTEM_PROP));
+        }
+        Path source = ddfHome.resolve(truststore);
+        Path destination = constructDestination(source, exportDirectory);
+        copyFile(source, destination);
+    }
+
+    private void exportCrl(Path exportDirectory) throws MigrationException {
+        Path encryptionPropertiesFile = ddfHome.resolve(FILE_CONTAINING_CRL_LOCATION);
+        try (Stream<String> lines = getStream(encryptionPropertiesFile)) {
+            String matchingCrlLine = lines.filter(line -> StringUtils.contains(line, CRL_PROP_KEY))
+                    .findFirst().get();
+            Path source = ddfHome.resolve(Paths.get(StringUtils.split(matchingCrlLine, "=")[1]));
+            Path destination = constructDestination(source, exportDirectory);
+            copyFile(source, destination);
+        } catch (IOException e) {
+            throw new MigrationException(String.format("Unable to read [%s].",
+                    encryptionPropertiesFile), e);
+        }
+    }
+
+    private void exportSecurityDirectory(Path exportDirectory) {
+        try {
+            Path source = ddfHome.resolve(Paths.get(SECURITY_DIRECTORY));
+            Path destination = constructDestination(source, exportDirectory);
+            copyDirectory(source, destination);
+        } catch (MigrationException e) {
+            LOGGER.warn(String.format("Unable to copy %s. It doesn't exist.", SECURITY_DIRECTORY),
+                    e);
+        }
+    }
+
+    private void verifyWithinDdfHome(Path path) throws MigrationException {
+        if (!StringUtils.startsWith(path.normalize().toString(), ddfHome.toString())) {
```

Comment:
We probably want to use `toRealPath()` instead of `normalize` on both `path` and `ddfHome` to make sure we don't run into problems with symbolic links and such. And maybe use `Path.startsWith()` instead of `StringUtils.startsWith()`.